src/node_internals.h

```diff
@@ -83,6 +84,11 @@ extern bool config_pending_deprecation;
 // Tells whether it is safe to call v8::Isolate::GetCurrent().
 extern bool v8_initialized;
 
+// Contains initial debug options.
+// Set in node.cc.
+// Used in node_config.cc.
+extern node::DebugOptions debug_options;
```

Comment:
This is where I'd put it. I can see how it has wider scope than the two files that will use it, but putting it in a header guarantees consistency between the place of definition and place of use.